src/extractor/guidance/motorway_handler.cpp

```diff
@@ -120,8 +120,16 @@ Intersection MotorwayHandler::fromMotorway(const EdgeID via_eid, Intersection in
         for (const auto &road : intersection)
         {
             const auto &out_data = node_based_graph.GetEdgeData(road.turn.eid);
-            if (road.turn.angle != 0 && in_data.name_id == out_data.name_id &&
-                in_data.name_id != EMPTY_NAMEID && isMotorwayClass(road.turn.eid, node_based_graph))
+
+            const auto same_name =
```

Comment:
see my comment above. The `out_data.name_id` cannot be empty here to preserve the handling based on transitivity.